frontend/src/auth/SignupForm.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useContext } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import './SignupForm.css'
+import { UserContext } from '../UserContext.js';
+
+const SignupForm = () => {
```

Comment:
This component is very similar to your LoginForm component. I'd look into how you can use one component instead and use props to make them different!